### PR TITLE
Rework README structure and introduction

### PR DESCRIPTION
**Pull request by HeavyWombat**
Add `pina-golada` logo as pure markdown.

Add badge for releses details.

Rework the introduction section of the README.

Restructure the main sections in the README.

Add two more sub-sections to the contribution sections for
Git Pre-Commit hooks and copy'n'paste Docker run command.

Fix markdown lint issues.

Make background transparent in logo file.

Create smaller version of the logo for the README.

### CODE REVIEW

**Comment by HeavyWombat** [comment]
This should fix #44 and #42.

File: README.md [hunk 1]
```diff
@@ -1,19 +1,66 @@
...
+
+import (
+  "github.com/homeport/pina-golada/pkg/files"
+)
+
+// Provider is the provider instance to access the assets framework
+var Provider ProviderInterface
+
+// ProviderInterface is the interface that provides the assets
+// @pgl(package=assets&injector=Provider)
```
[the remark this answers is not in the document]
Comment (LynxPlay101) [reply README.md]: The correct annotation would simply be 

`@pgl(injector=Provider)`

File: README.md [hunk 2]
```diff
@@ -1,19 +1,66 @@
...
 [![GoDoc](https://godoc.org/github.com/homeport/pina-golada?status.svg)](https://godoc.org/github.com/homeport/pina-golada)
+[![Release](https://img.shields.io/github/release/homeport/pina-golada.svg)](https://github.com/homeport/pina-golada/releases/latest)
+
+![pina-golada](.docs/logo-small.png?raw=true "Pina-Golada logo by Michael Schepanske")
 
 ## Introducing the pina-golada
 
-Pina-Golada is a tool to automatically generate asset providers that implement custom interfaces.
-The tool's main propose is to hide ugly and blown up asset provider behind interfaces.  
+Pina-Golada is a tool and framework to package assets (files and directories) into an executable binary. To be precise, it is for executables that are created by the Go compiler. For this, `pina-golada` scans your Go source files for well known annotations and creates Go source code that contains the referenced assets. In your program, you are able to access the assets in memory or write them to disk.
```
[the remark this answers is not in the document]
Comment (LynxPlay101) [reply README.md]: From the github repository description: 
"pina-golada is a simple asset tool for go, which generates interface implementations that provide files/folders in the final build, without rendering them in the source code themselve"

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (HeavyWombat) [reply README.md]: Thanks for the hint. Will be fixed.

File: README.md [hunk 3]
```diff
@@ -1,19 +1,66 @@
...
+type ProviderInterface interface {
+  
+  // GetConfigFile returns an in-memory version of the asset
+  // @pgl(asset=/assets/default-config.yaml&compressor=tar)
+  GetConfigFile() (dir files.Directory , e error)
+}
+```
+
+Prior to the build, run `pina-golada generate` to create a generated Go source file, which contains your assets. To clean up any generated file, use `pina-golada cleanup`.
+
```
[the remark this answers is not in the document]
Comment (HeavyWombat) [reply README.md]: Good point.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (HeavyWombat) [reply README.md]: Let me try to add your last sentence in the introduction text.

**Comment by LynxPlay101** [comment]
This fixed #40

**Pull request merged**